scripts/deleteAndLoadSnapsets.js

```diff
@@ -0,0 +1,523 @@
+ * @param {Object} options - Object that contains the sets of Prefs Safes and
+ *                           their keys.
+ * @return {Object} - The number of snapsets and gpiiKeys deleted.
+ */
+gpii.dataLoader.logSnapsetDeletion = function (responseString, options) {
+    fluid.log(  "Deleted " +
+                options.snapsetPrefsSafes.length + " Prefs Safes and " +
+                options.gpiiKeys.length + " associated GPII Keys, "
+    );
+    return {
+        snapsets: options.snapsetPrefsSafes.length,
+        gpiiKeys: options.gpiiKeys.length
+    };
+};
+
+/*
+ * Create the step that deletes, in batch, the current snapset Prefs Safes and
+ * their associated GPII keys.
+ * @param {Promise} previousStep - Promise from a previous step whose fulfillment
+ *                                 triggers the bulk delete request.
+ * @param {Object} options - Object that contains the records to be deleted.
+ * @return {Promise} - The promise that resolves the deletion.
+ */
+gpii.dataLoader.createBatchDeleteStep = function (options) {
+    var togo = fluid.promise();
+    var response = gpii.dataLoader.createResponseHandler(
+        gpii.dataLoader.logSnapsetDeletion,
+        options,
+        togo
+    );
+    var batchDeleteRequest = gpii.dataLoader.configureBatchDelete(response, options);
+    batchDeleteRequest.end();
+    return togo;
+};
+
+/*
+ * Log the uploading of all the snapset and user Prefs Safes.
+ * @param {String} responseString - Response from the database (ignored)
+ * @param {Object} options - Object that contains the directories from which
+ *                           the Prefs Safes were loaded.
+ */
+gpii.dataLoader.logSnapsetsUpload = function (responseString, options) {
+    fluid.log("Bulk loading of build data from '" + options.buildDataDir + "'");
+    return "Uploaded latest snapsets preferences";
+};
+
+/*
+ * Create the step that uploads, in batch, the new snapset Prefs Safes, and
+ * their associated GPII keys.
+ * @param {Object} options - Object that has the paths to the directory that
+ *                           contains the latest snapsets.
+ * @return {Promise} - A promise that resolves the upload.
+ */
+gpii.dataLoader.createBatchUploadStep = function (options) {
+    var togo = fluid.promise();
+    var buildData = gpii.dataLoader.getDataFromDirectory(options.buildDataDir);
+    var response = gpii.dataLoader.createResponseHandler(
+        gpii.dataLoader.logSnapsetsUpload,
+        options,
+        togo
+    );
+    var bulkUploadRequest = gpii.dataLoader.createPostRequest(buildData, response, options);
+    bulkUploadRequest.end();
+    return togo;
+};
+
+/*
+ * Create and execute the steps to update the database.
+ */
+gpii.dataLoader.orchestrate = function () {
+    var options = gpii.dataLoader.initOptions(process.argv);
+    gpii.dataLoader.loadStaticDataFromDisk(options);
+    var sequence = [
+        gpii.dataLoader.createStaticDataStep,
```

Comment:
There are quite a similarity with the implementation logic for these steps. Do you think if it's possible to create an utility function to share?